streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -67,6 +68,58 @@
     }
 
     @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        final List<Segment> searchSpace = keySchema.allSegments(segments);
+        
+        final Segment minSegment = keySchema.minSegment(segments);
+        final Segment maxSegment = keySchema.maxSegment(segments);
+        final Bytes minKey = minSegment.first().key;
+        final Bytes maxKey = maxSegment.last().key;
+        
+        final long minTimestamp = keySchema.segmentTimestamp(minKey);
+        final long maxTimestamp = keySchema.segmentTimestamp(maxKey);
+        final Bytes keyFrom = WindowStoreUtils.bytesKeyFromBinaryKey(minKey.get());
+        final Bytes keyTo = WindowStoreUtils.bytesKeyFromBinaryKey(maxKey.get());
+        
+        return new SegmentIterator(searchSpace.iterator(),
+                                   keySchema.hasNextCondition(keyFrom, keyTo, minTimestamp, maxTimestamp),
+                                   minKey, maxKey);
+    }
+    
+    @Override
+    public KeyValueIterator<Bytes, byte[]> fetchAll(final long timeFrom, final long timeTo) {
+        final List<Segment> searchSpace = keySchema.allSegments(segments);
+        long minTimestamp = Long.MAX_VALUE;
+        long maxTimestamp = 0;
+        Bytes keyFrom = null;
```

Comment:
As above in caching, can keyFrom just be `new Bytes(new byte[0])`?
If that is true then you could perhaps do something like:
```
final Segment lastSegment = segments.getSegmentForTimestamp(timeTo);
// find keyTo in lastSegment
```
That might help to avoid having to iterate over the entire window store here to find the min and max. we'd only need to iterate over at most one segment.